phoenicis-javafx/src/main/java/org/phoenicis/javafx/components/engine/skin/EnginesSidebarToggleGroupSkin.java

```diff
@@ -36,7 +36,7 @@ public EnginesSidebarToggleGroupSkin(EnginesSidebarToggleGroup control) {
     protected ToggleButton convertToToggleButton(EngineCategoryDTO category) {
         final ToggleButton categoryButton = createSidebarToggleButton(category.getName());
 
-        categoryButton.setId(String.format("%sButton", category.getName().toLowerCase()));
+        categoryButton.setId(String.format("%s-button", category.getName().toLowerCase().replace('.', '-')));
```

Comment:
I think it would be useful to add a method `buttonize`, which is responsible for the whole conversion of a string:

```java
public String buttonize(String pattern, String input) {
   return String.format(pattern, input.toLowerCase().replace('.', '-'))
}
```

Reply:
I thought about this but in which class?

Reply:
If this is only required in the sidebars I would recommend `SidebarToggleGroupBaseSkin`.
I also think it makes sense to make the method static.